core/transformer/src/IModelTransformer.ts

```diff
@@ -463,11 +473,75 @@ export class IModelTransformer extends IModelExportHandler {
 
   /** Initialize the source to target Element mapping from ExternalSourceAspects in the target iModel.
    * @note This method is called from [[processChanges]] and [[processAll]], so it only needs to be called directly when processing a subset of an iModel.
+   * @note Passing an [[InitFromExternalSourceAspectsArgs]] is required when processing changes, to remap any elements that may have been deleted.
+   *       You must await the returned promise as well in this case. The synchronous behavior has not changed but is deprecated and won't process everything.
    */
-  public initFromExternalSourceAspects(): void {
+  public initFromExternalSourceAspects(args?: InitFromExternalSourceAspectsArgs): Promise<void>;
```

Comment:
Breaking change of return type from `void` to `Promise<void>` is ok because IModelTransformer is beta but please document in NextVersion.md.

Reply:
Oh weird, your overload returns `void | Promise<void>`. Does that affect existing callers?

Reply:
oh yes will add to NextVersoin.md, but it's not breaking, it still has an overload of void, it's just deprecated.
I did consider using beta to not have an overload but